resources.Groups.department.ReferenceExample.ps1

```diff
@@ -24,102 +131,206 @@ catch {
-            # $ADGroupDescription = ("$adGroupDescriptionPrefix" + "$($resource.Name)" + "$adGroupDescriptionSuffix")
-
-            # Example: Custom field (custom fields consists of only one attribute, no object with multiple attributes present!)
-            # $ADGroupName = ("$adGroupNamePrefix" + "$($resource)" + "$adGroupNameSuffix")
-            # $ADGroupDescription = ("$adGroupDescriptionPrefix" + "$($resource)" + "$adGroupDescriptionSuffix")
-
-            $ADGroupName = Get-ADSanitizeGroupName -Name $ADGroupName
-
-            $ADGroupParams = @{
-                Name           = $ADGroupName
-                SamAccountName = $ADGroupName
-                GroupCategory  = "Security"
-                GroupScope     = "Global"
-                DisplayName    = $ADGroupName
-                Path           = $path
-                Description    = $ADGroupDescription
+        Write-Verbose "Checking $($resource)"
+
+        # Check if required fields are available in resource object
+        $incompleteResource = $false
+        $missingFields = [System.Collections.ArrayList]@()
+        foreach ($requiredField in $requiredFields) {
+            if ($requiredField -notin $resource.PsObject.Properties.Name) {
+                $incompleteResource = $true
+                [void]$missingFields.Add($requiredField)
+                if ($debug -eq $true) { Write-Warning "Resource object is missing required field [$requiredField]" }
+            }
+
+            if ([String]::IsNullOrEmpty($resource.$requiredField)) {
+                $incompleteResource = $true
+                [void]$missingFields.Add($requiredField)
+                if ($debug -eq $true) { Write-Warning "Resource object has a null or empty value for required field [$requiredField]" }
             }
+        }
+
+        if (-Not($incompleteResource -eq $True)) {
+            try {
+                # The names of security principal objects can contain all Unicode characters except the special LDAP characters defined in RFC 2253.
+                # This list of special characters includes: a leading space a trailing space and any of the following characters: # , + " \ < > 
+                # A group account cannot consist solely of numbers, periods (.), or spaces. Any leading periods or spaces are cropped.
+                # https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2003/cc776019(v=ws.10)?redirectedfrom=MSDN
+                # https://www.ietf.org/rfc/rfc2253.txt
 
-            $distinguishedName = "CN=$($ADGroupParams.Name),$($ADGroupParams.Path)"
-            $groupExists = [bool](Get-ADGroup -Filter { DistinguishedName -eq $distinguishedName } -Server $pdc)
+                # Example: Department (department differs from other objects as the property for the name is "DisplayName", not "Name")
+                $ADGroupName = ("$adGroupNamePrefix" + "$($resource.DisplayName)" + "$adGroupNameSuffix")
```

Comment:
Probeer te sturen in je standaardcode om de externalId van de functie/afdeling of gecombineeerde waarde in de CN en samaccountname te zetten, en displayname in de description van de groep. Anders loop je regelmatig uit de max van 64 chars bij het aanmaken van groepen